lib/matplotlib/backends/backend_cairo.py

```diff
@@ -132,8 +132,18 @@ def _fill_and_stroke (self, ctx, fill_c, alpha, alpha_overrides):
             ctx.restore()
         ctx.stroke()
 
-    @staticmethod
-    def convert_path(ctx, path, transform):
+    def convert_path(self, ctx, path, transform):
+        # First clip the path to the actual rendering extents
+        points = path.get_extents().get_points()
+        width = points[1][0] - points[0][0]
+        height = points[1][1] - points[0][1]
+        wfactor = width / self.width
+        hfactor = height / self.height
+
+        extents = ctx.clip_extents()
+        clip_rect = (extents[0] * wfactor, extents[1] * hfactor, extents[2] * wfactor, extents[3] * hfactor)
```

Comment:
OK, will change once you've acked my other comment, I think amending commits and pushing kills line notes